applications/fishing-map/src/features/map/map.selectors.ts

```diff
@@ -111,40 +110,49 @@ export const getGeneratorsConfig = createSelector(
       generatorsConfig.push(mergedLayer)
     }
 
-    // Collect track generators and inject highligtedTime
-    generatorsConfig = generatorsConfig.map((generator) => {
-      const isTrack = generator.config?.type === Generators.Type.Track
-      if (!isTrack) return generator
-      return {
-        ...generator,
-        config: {
-          ...generator.config,
-          highlightedTime,
-        },
-      }
-    })
-
     generatorsConfig = generatorsConfig.flatMap((dataview) => {
-      const config: DataviewConfig = { ...dataview.config }
-      // Try to retrieve resource if it exists
-      let data
-      const { url } = resolveDataviewDatasetResource(dataview, TRACKS_DATASET_TYPE)
-      if (url && resources[url]) {
-        data = resources[url].data
+      const generator: GeneratorDataviewConfig = {
+        id: dataview.id,
+        ...dataview.config,
       }
 
-      const generator: any = {
-        ...config,
-        id: dataview.id,
-        ...(data && { data }),
+      if (dataview.config?.type === Generators.Type.Track) {
+        // Inject highligtedTime
+        generator.config = {
+          ...generator.config,
+          highlightedTime,
+        }
+        // Try to retrieve resource if it exists
+        const { url } = resolveDataviewDatasetResource(dataview, { type: TRACKS_DATASET_TYPE })
+        if (url && resources[url]) {
+          generator.data = resources[url].data
+        }
       }
-      // TODO: remove UserContext and use Context
-      if (dataview.config?.type === Generators.Type.UserContext) {
-        const { url } = resolveDataviewDatasetResource(dataview, USER_CONTEXT_TYPE)
-        if (url) {
-          generator.tilesUrl = url
+
+      if (dataview.config?.type === Generators.Type.Context) {
+        if (Array.isArray(dataview.config.layers)) {
+          const tilesUrls = dataview.config.layers?.flatMap(({ id, dataset }) => {
+            const { url } = resolveDataviewDatasetResource(dataview, { id: dataset })
+            if (!url) return []
+            return { id, tilesUrl: url }
+          })
+          // Duplicated generators when context dataview have multiple layers
+          return tilesUrls.map(({ id, tilesUrl }) => ({
+            ...generator,
+            id: dataview.id + id,
```

Comment:
Shouldn't w avoid concatenating strings with `+` ? We used to have an eslint rule for that no?

Reply:
(personally I prefer `[dataview.id, id].join('_')`

Reply:
Bought 💰